pyrenew/distributions/censorednormal.py

```diff
@@ -0,0 +1,107 @@
+# numpydoc ignore=GL08
+
+import jax
+import jax.numpy as jnp
+import numpyro
+import numpyro.util
+from numpyro.distributions import constraints
+from numpyro.distributions.util import promote_shapes, validate_sample
+
+
+class CensoredNormal(numpyro.distributions.Distribution):
+    """
+    Censored normal distribution under which samples
+    are truncated to lie within a specified interval.
+    This implementation is adapted from
+    https://github.com/dylanhmorris/host-viral-determinants/blob/main/src/distributions.py
+    """
+
+    arg_constraints = {"loc": constraints.real, "scale": constraints.positive}
+    support = constraints.real
```

Comment:
This is my fault since it's inexact in the reference implementation, but we should have the support specified dynamically at initialization, since strictly speaking it depends on the `lower_limit` and `upper_limit` values.

I'd follow the pattern used for the `Uniform`
https://num.pyro.ai/en/stable/_modules/numpyro/distributions/continuous.html#Uniform

1. define a `self._support` attribute at initialization
2. Define a property method
```python
    @constraints.dependent_property(is_discrete=False, event_dim=0)
    def support(self):
        return self._support
```

We probably also want to declare `lower_limit`, `upper_limit`, and `support` as `pytree_data_fields`.

Reply:
> We probably also want to declare `lower_limit`, `upper_limit`, and `support` as `pytree_data_fields`.

shouldn't `loc` and `scale` also be declared as `pytree_data_fields` if we are using it?